doc/rds-day-2.md

```diff
@@ -0,0 +1,184 @@
+
+In Parameter Groups, there are two types of parameters: dynamic and static. Dynamic parameters do not require a restart for their values to be applied to the running instance / cluster.  Static parameters require a restart for their values to be applied. Additionally, dynamic parameters support specifying how changes to them are applied. When `immediate` is specified the changes to dynamic parameters are applied immediately. When `pending-reboot` is specified, the changes to dynamic parameters are applied during next restart or during the next maintenance window, whichever is earlier. 
+
+Since static parameters do not support `immediate` apply option, specifying this in your composition could lead to some unexpected errors. Therefore, extra care should be taken when exposing this resource to your end users. End users may not be aware of underlying engine specifications.
+
+Summarizing everything above effectively means there are a few general approaches to managing RDS configuration changes. 
+1. You want to ensure that parameter group values in the running cluster / instance match what is defined in your Git repository with no delay. The only certain way do this is by restarting the cluster/ instance during the reconciliation process.
+2. You can wait for parameter group changes to be applied during the next maintenance window. This means you may need to wait maximum 7 days for the changes to be applied. 
+3. The change does not have to be applied immediately but it needs to happen sooner than 7 days. This requires a separate workflow to restart cluster / instance.
+4. Use the RDS Blue Green deployment feature.
+
+For reference, problems encountered during parameter group updates in ACK and Terraform are discussed in [this issue](https://github.com/aws-controllers-k8s/community/issues/869) and [this blog post](https://tech.instacart.com/terraforming-rds-part-3-9d81a7e2047f). 
+
+## Solutions
+
+### Considerations
+As of writing this doc, there are 9 fields that require a restart to take effect when using a single RDS instance. There are 3 fields that require a restart when using multi-AZ instances. Unfortunately there is no native way to get these fields programmatically.
+
+There are 188 static parameters in `mysql8.0` family, and similar number of them are in other parameter group families as well. You can get a list of static parameters by using the `aws rds describe-engine-default-parameters` command. 
+
+These fields and parameters need to be stored for use by whatever check mechanism you choose, and they need to be updated regularly.
```

Comment:
If I understand well, those parameters can change with time? ACK stores them [in-memory](https://github.com/aws-controllers-k8s/rds-controller/blob/main/pkg/resource/db_parameter_group/hooks.go#L551-L563) and we don't have a mechanism that syncs them automatically. Should we consider a regular cache sync, if yes, what's a reasonable frequency? or maybe it's worth it to just query all the parameter groups every time we need them?

Reply:
Yes. My understanding is that they could change especially with underlying engine updates. However, I think it happens very infrequently so what you did in ACK is perfectly acceptable. I'd say keep it simple now and refactor when a need for it arises.